src/device/intrinsics/math.jl

```diff
@@ -127,16 +171,65 @@ end
 
 @device_override Base.exp(x::Float64) = ccall("extern __nv_exp", llvmcall, Cdouble, (Cdouble,), x)
 @device_override Base.exp(x::Float32) = ccall("extern __nv_expf", llvmcall, Cfloat, (Cfloat,), x)
+@device_override function Base.exp(h::Float16)
+    # perform computation in Float32 domain
+    f = Float32(h)
+    f = fma(f, reinterpret(Float32, 0x3fb8aa3b), reinterpret(Float32, Base.sign_mask(Float32)))
```

Comment:
```
julia> reinterpret(UInt32, log2(Float32(ℯ)))
0x3fb8aa3b
```

and:

```
julia> reinterpret(Float32, Base.sign_mask(Float32))
-0.0f0
```

We probably can't rely on the constant evaluation of this, but this code is essentially: `f *= log2(Float32(ℯ))`